api/src/main/java/com/ichi2/anki/api/AddContentApi.java

```diff
@@ -151,6 +158,47 @@ public int addNotes(long modelId, long deckId, List<String[]> fieldsList, List<S
         return getCompat().insertNotes(deckId, newNoteValuesList.toArray(new ContentValues[newNoteValuesList.size()]));
     }
 
+    /**
+     * Add a media file to AnkiDroid's media collection. You would likely supply this uri through a FileProvider, and
+     * then set FLAG_GRANT_READ_URI_PERMISSION using something like:
+     * getContext().grantUriPermission("com.ichi2.anki", uri, Intent.FLAG_GRANT_READ_URI_PERMISSION) and then afterward
+     * remove the permission with getContext().revokePermission(uri, Intent.FLAG_GRAN_READ_URI_PERMISSION)
+     * @param fileUri   Uri for the file to be added.
+     * @param preferredName String that will be added to the beginning of the file name in the media collection
+     * @param mimeType  String indicating the mimeType of the media. Accepts "audio" or "image"
+     * @return the correctly formatted String for the media file to be placed in the desired field of a Card.
+     */
+    public String addMediaFromUri(Uri fileUri, String preferredName, String mimeType) {
```

Comment:
Should mark it up with expectations on what can be null and what can't both in docs and as annotations, to guard against crashes - those are common sources of crashes for us

```suggestion
     * @param fileUri  Uri for the file to be added, required.
     * @param preferredName String that will be added to the beginning of the file name in the media collection, required
     * @param mimeType  String indicating the mimeType of the media. Accepts "audio" or "image", required
     * @return the correctly formatted String for the media file to be placed in the desired field of a Card or null if unsuccessful.
     */
    public @Nullable String addMediaFromUri(@NonNull Uri fileUri, @NonNull String preferredName, @NonNull String mimeType) {
```

Reply:
When I apply the annotations it shows a warning and gives me some options: 
"Add 'annotations' to classpath", and
"Add dependency on androidx.annotation:annotation"
Both of these edit the build.gradle file for api so I'm wondering what I should do here?

Reply:
Oh interesting! The API module hasn't gotten much love in a while and I guess we just hadn't used annotations there yet. Great question, thank you for asking. Adding those to the build.gradle is the right thing to do